R/splitModels.R

```diff
@@ -1,110 +1,79 @@
-    if(!is.null(qMatrix))       {qMatrix      <- checkQmatrixConsistency (qMatrix)}
-    if(!is.null(person.groups)) {person.groups<- checkPersonGroupsConsistency (person.groups)}
-    mcPackage <- match.arg(mcPackage)
-    # Funktion: person.groups nach person.grouping
-		pg2pgr <- function ( x , nam ) {
-				d <- x[,1,drop=FALSE]
-				eval ( parse ( text = paste0 ( "d$'" , nam , "' <- 1 " ) ) )
-				return ( d )
-		}
-		# wenn kein data.frame, dann ignorieren
-		if ( !is.null ( qMatrix ) & !is.data.frame ( qMatrix ) ) {
-				qMatrix <- NULL
-				warning ( paste0 ( "splitModels: qMatrix is not a data.frame and will be ignored." ) , call. = FALSE )
-		}
-		if ( !is.null ( person.groups ) & !is.data.frame ( person.groups ) ) {
-				person.groups <- NULL
-				warning ( paste0 ( "splitModels: person.groups is not a data.frame and will be ignored." ) , call. = FALSE )
-		}
-		# wenn keine Spalten / Zeilen dann NULL
-		if ( !is.null ( qMatrix ) ) {
-				if ( nrow ( qMatrix ) %in% 0 | ncol ( qMatrix ) %in% 0 ) {
-						warning ( "splitModels: check qMatrix" , call. = FALSE )
-						qMatrix <- NULL
-				}
-		}
-		if ( !is.null ( person.groups ) ) {
-				if ( nrow ( person.groups ) %in% 0 | ncol ( person.groups ) %in% 0 ) {
-						warning ( "splitModels: check person.groups" , call. = FALSE )
-						person.groups <- NULL
-				}
-		}
-		# Dimensionen in qMatrix duerfen nur 0/1 haben
-		if ( !is.null ( qMatrix ) ) {
-				if ( ncol ( qMatrix ) > 1 ) {
-						not01 <- ! sapply ( qMatrix[,-1,drop=FALSE] , function ( x ) all ( x %in% c(0,1) ) )
+splitModels <- function(qMatrix = NULL, person.groups = NULL, split = c("qMatrix", "person.groups"),
+                        add = NULL, cross = NULL, all.persons = TRUE, all.persons.lab = "all",
+                        person.split.depth = 0:length(person.groups[,-1,drop=FALSE]),
+                        full.model.names = TRUE, model.name.elements = c("dim", "group", "cross"),
+                        include.var.name = FALSE, env = FALSE, nCores=NULL, mcPackage = c("future", "parallel"),
+                        GBcore=NULL, verbose = TRUE){
+
+### checking/asserting the arguments -------------------------------------------
+
+  if(!is.null(qMatrix)){
+    qMatrix <- checkQmatrixConsistency(qMatrix)}
+  if(!is.null(person.groups)){
+    person.groups<- checkPersonGroupsConsistency(person.groups)}
+  mcPackage <- match.arg(mcPackage)
+  # Funktion: person.groups nach person.grouping
+  pg2pgr <- function(x, nam){
+    d <- x[, 1, drop = FALSE]
+    eval(parse(text = paste0("d$'", nam, "' <- 1 ")))
+    return(d)
+  }
+
+  lapply(c(all.persons, full.model.names, include.var.name, env, verbose), checkmate::assert_logical, len = 1)
+  checkmate::assert_subset(split, choices = c("qMatrix" , "person.groups"))
+  checkmate::assert_subset(model.name.elements, choices = c("dim", "group", "add", "cross"))

```

Comment:
In the function head, `c("dim", "group", "cross")` is allowed for `model.name.elements`, here, `c("dim", "group", "add", "cross")` is allowed? (seem to be a mismatch)